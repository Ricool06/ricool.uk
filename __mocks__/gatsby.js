import React from 'react';

const gatsby = jest.requireActual('gatsby');

module.exports = {
  ...gatsby,
  graphql: jest.fn(queryString => queryString),
  Link: jest.fn().mockImplementation(({ to, ...rest }) => React.createElement('a', {
    ...rest,
    href: to,
  })),
  StaticQuery: jest.fn(),
};
