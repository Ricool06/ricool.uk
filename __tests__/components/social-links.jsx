import React from 'react';

import { StaticQuery } from 'gatsby';
import renderer from 'react-test-renderer';
import { render, cleanup } from 'react-testing-library';
import SocialLinks, { PureSocialLinks } from '../../src/components/social-links';

describe('SocialLinks', () => {
  const data = {
    allSocialJson: {
      edges: [
        {
          node: {
            url: 'https://github.com/Ricool06',
            image: {
              childImageSharp: {
                id: 'af470c94-725d-50fd-937f-2d9729f0d809',
                fluid: {
                  originalName: 'github.png',
                  aspectRatio: 1,
                  src: '/static/472739dfb5857b1f659f4c4c6b4568d0/d7cee/github.png',
                  srcSet: '/static/472739dfb5857b1f659f4c4c6b4568d0/d7cee/github.png',
                  sizes: '(max-width: 300px) 100vw, 300px',
                },
              },
            },
          },
        },
        {
          node: {
            url: 'https://twitter.com/Ricool06',
            image: {
              childImageSharp: {
                id: '65e9f256-76cb-5750-9963-fb2d50c89627',
                fluid: {
                  originalName: 'twitter.png',
                  aspectRatio: 1,
                  src: '/static/d73d21cfc1df81e923d62e220e790022/045aa/twitter.png',
                  srcSet: '/static/d73d21cfc1df81e923d62e220e790022/045aa/twitter.png',
                  sizes: '(max-width: 300px) 100vw, 300px',
                },
              },
            },
          },
        },
      ],
    },
  };

  afterEach(cleanup);

  it('should render all images with links passed to it', () => {
    const { container } = render(<PureSocialLinks data={data} />);
    const images = container.querySelectorAll('img');
    const links = container.querySelectorAll('a');

    expect(links).toHaveLength(2);
    expect(images).toHaveLength(2);

    images.forEach((image, key) => expect(image).toHaveAttribute('src', data.allSocialJson.edges[key].node.image.childImageSharp.fluid.src));
    links.forEach((link, key) => expect(link).toHaveAttribute('href', data.allSocialJson.edges[key].node.url));
  });


  it('should match the snapshot', () => {
    StaticQuery.mockImplementation(({ render: mockRender }) => mockRender(data));

    const jsonSnapshot = renderer.create(<SocialLinks />).toJSON();
    expect(jsonSnapshot).toMatchSnapshot();
  });
});
