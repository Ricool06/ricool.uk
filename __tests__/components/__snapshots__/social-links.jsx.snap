// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks should match the snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "marginLeft": 1,
        "width": 50,
      }
    }
  >
    <a
      href="https://github.com/Ricool06"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "100%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 300px) 100vw, 300px"
            srcSet="/static/472739dfb5857b1f659f4c4c6b4568d0/d7cee/github.png"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 300px) 100vw, 300px"
            src="/static/472739dfb5857b1f659f4c4c6b4568d0/d7cee/github.png"
            srcSet="/static/472739dfb5857b1f659f4c4c6b4568d0/d7cee/github.png"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/472739dfb5857b1f659f4c4c6b4568d0/d7cee/github.png\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" srcset=\\"/static/472739dfb5857b1f659f4c4c6b4568d0/d7cee/github.png\\" src=\\"/static/472739dfb5857b1f659f4c4c6b4568d0/d7cee/github.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </a>
  </div>,
  <div
    style={
      Object {
        "marginLeft": 1,
        "width": 50,
      }
    }
  >
    <a
      href="https://twitter.com/Ricool06"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "100%",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 300px) 100vw, 300px"
            srcSet="/static/d73d21cfc1df81e923d62e220e790022/045aa/twitter.png"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 300px) 100vw, 300px"
            src="/static/d73d21cfc1df81e923d62e220e790022/045aa/twitter.png"
            srcSet="/static/d73d21cfc1df81e923d62e220e790022/045aa/twitter.png"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/d73d21cfc1df81e923d62e220e790022/045aa/twitter.png\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" srcset=\\"/static/d73d21cfc1df81e923d62e220e790022/045aa/twitter.png\\" src=\\"/static/d73d21cfc1df81e923d62e220e790022/045aa/twitter.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </a>
  </div>,
]
`;
